### ISSUE TITLE
cyber_recorder: command not found

### ISSUE THREAD

**Issue by deepaktalwardt** [ApolloAuto/apollo#6764]
- **OS Platform and Distribution (e.g., Linux Ubuntu 14.04)**: Ubuntu 16.04 LTS
- **Apollo installed from (source or binary)**: source
- **Apollo version (1.0, 1.5, 2.0, 2.5, 3.0)**: 3.5

### Steps to reproduce the issue:
I do not have access to hardware which is why I followed the instructions to install without hardware here: https://github.com/ApolloAuto/apollo/tree/master/docs/demo_guide

Everything works fine until the cyber_recorder step. This is the output I get:
`root@in_dev_docker:/apollo# bash scripts/bootstrap.sh` 
`nohup: appending output to 'nohup.out'`
`Launched module monitor.`
`nohup: appending output to 'nohup.out'`
`Launched module dreamview.`
`Dreamview is running at http://localhost:8888`
`root@in_dev_docker:/apollo# cyber_recorder play -f docs/demo_guide/demo_3.5.record --loop`
`bash: cyber_recorder: command not found`

### Supporting materials (screenshots, command lines, code/script snippets):
![image](https://user-images.githubusercontent.com/7192506/52378455-8fb58680-2a1c-11e9-85a1-ce79c0133c62.png)

This is what http://localhost:8888/ looks like.

Please help me resolve this. Thank you!

**Comment by jilinzhou** [comment]
1. Check whether the binary cyber_recorder is existed. It should be under
`/apollo/bazel-bin/cyber/tools/cyber_recorder/`

2. Verify the environment variable $PATH
`>>echo $PATH`
The path should include _/apollo/bazel-bin/cyber/tools/cyber_recorder/_. If not, source cyber/setup.bash

**Comment by FreshmanD** [comment]
You can source the setup.bash file again at the beginning, using 
source /your-path-to-apollo-install-dir/cyber/setup.bash

**Comment by deepaktalwardt** [comment]
Thank you @jilinzhou and @FreshmanD. I sourced the setup.bash file within docker and tried running however, it still doesn't work. Here is the screenshot.
![image](https://user-images.githubusercontent.com/7192506/52547460-bce79900-2d7c-11e9-9412-5fa51160eaf7.png)

But I'm having another issue now. I tried running the "bash apollo.sh build" command again, but now I get a permission denied error. I don't believe I got this before. I did run the additional steps for docker in ubuntu to be able to use it without sudo but it doesn't seem to work. If you could help me sort out the permissions that would be great as well. 
![image](https://user-images.githubusercontent.com/7192506/52547506-1f409980-2d7d-11e9-96e6-1fed03873ebd.png)

**Comment by FreshmanD** [comment]
Do u use GPU to build the system？

**Comment by deepaktalwardt** [comment]
Yeah @FreshmanD, I used the regular build command. I have an NVIDIA GTX 1050 with the correct drivers. Should I try building with the cpu command?
![image](https://user-images.githubusercontent.com/7192506/52550123-9a10b100-2d8b-11e9-8514-a28144c062bf.png)

**Comment by FreshmanD** [comment]
I think u should clean the doctor and restart it, if that still doesn't work, u can use VS to build it.

**Comment by deepaktalwardt** [comment]
Can you provide me with instructions to clean docker? Thank you!

**Comment by FreshmanD** [comment]
$ docker container stop $(docker container ls -a -q) && docker system prune --all --force --volumns

**Comment by FreshmanD** [comment]
Remember to check whether that error package is not really existed or not

**Comment by deepaktalwardt** [comment]
> Remember to check whether that error package is not really existed or not

I checked and the module did exist. Which is why it is strange. Thanks, I'm trying to clean the docker and retrying the build.

**Comment by sbanga16k** [comment]
@deepaktalwardt - If you have permission denied error, try running:
sudo bash dev_start.sh & dev_into.sh
Then source cyber/setup.bash like was mentioned earlier. 
It probably should work now

**Comment by deepaktalwardt** [comment]
I needed to provide all permissions to my user before starting the docker and running the build. That fixed the issue. Thank you all.

**Comment by jewes** [comment]
> 1. Check whether the binary cyber_recorder is existed. It should be under
>    `/apollo/bazel-bin/cyber/tools/cyber_recorder/`
> 2. Verify the environment variable $PATH
>    `>>echo $PATH`
>    The path should include _/apollo/bazel-bin/cyber/tools/cyber_recorder/_. If not, source cyber/setup.bash

It works.

**Comment by ZefuLin** [comment]
first vim cyber/setup.bash
if you can see

#! /usr/bin/env bash
TOP_DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd -P)"
source ${TOP_DIR}/scripts/apollo.bashrc
 
export APOLLO_BAZEL_DIST_DIR="${APOLLO_CACHE_DIR}/distdir"
export CYBER_PATH="${APOLLO_ROOT_DIR}/cyber"
 
bazel_bin_path="${APOLLO_ROOT_DIR}/bazel-bin"
mainboard_path="${bazel_bin_path}/cyber/mainboard"
cyber_tool_path="${bazel_bin_path}/cyber/tools"
recorder_path="${cyber_tool_path}/cyber_recorder"
launch_path="${cyber_tool_path}/cyber_launch"
channel_path="${cyber_tool_path}/cyber_channel"
node_path="${cyber_tool_path}/cyber_node"
service_path="${cyber_tool_path}/cyber_service"
monitor_path="${cyber_tool_path}/cyber_monitor"
visualizer_path="${bazel_bin_path}/modules/tools/visualizer"

Then:
bazel build cyber/tools/cyber_recorder/...

I solved the same problem by using the above steps!
